gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,532 @@
+#! @Description
+#! Set the <A>colour</A> of the <A>i</A>th inner circle (following the face numbering).
+#! @Returns the updated print record.
+#! @Arguments surface, i, colour, printRecord
+DeclareOperation( "SetCircleColour", [IsPolygonalComplex and IsNotEdgeRamified, IsCyclotomic, IsString, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup GetCircleColour
+#! @Description
+#! Get the colour of the <A>i</A>th inner circle (following the face numbering).
+#! @Returns the colour.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetCircleColour", [IsPolygonalComplex and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @Section Normales of Inner Circles
+#! @SectionLabel NormalesInnerCirclesAnimating
+#!
+
+# TODO add examples in this sections
+
+#! The normal vectors of inner circles can also be (de-)activated in the familiar way.
+#! If the are active, they have the colour of the corresponding inner circle.
+
+#! @BeginGroup ActivateNormalOfInnerCircles
+#! @Description
+#! Activate all normals innerer circles.
+#! @Returns the updated print record.
+#! @Arguments surface, printRecord
+DeclareOperation( "ActivateNormalOfInnerCircles", [IsPolygonalComplex and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup DeactivateNormalOfInnerCircles
+#! @Description
+#! Deactivate all normals of innerer circles.
+#! @Returns the updated print record.
+#! @Arguments surface, printRecord
+DeclareOperation( "DeactivateNormalOfInnerCircles", [IsPolygonalComplex and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup ActivateNormalOfInnerCircle
+#! @Description
+#! Activate the normal of the <A>i</A>th inner circle (following the face numbering).
+#! @Returns the updated print record.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "ActivateNormalOfInnerCircle", [IsPolygonalComplex and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup DeactivateNormalOfInnerCircle
+#! @Description
+#! Deactivate the normal of the <A>i</A>th inner circle (following the face numbering).
+#! @Returns the updated print record.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "DeactivateNormalOfInnerCircle", [IsPolygonalComplex and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup IsNormalOfInnerCircleActive
+#! @Description
+#! Test whether the normal of the <A>i</A>th inner circle is active (following the face numbering).
+#! @Returns a boolean.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "IsNormalOfInnerCircleActive", [IsPolygonalComplex and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @Section Main function <K>DrawSurfaceToJavaScript</K>
+#! @SectionLabel MainAnimating
+#!
+
+#! @BeginGroup DrawSurfaceToJavaScript
+#! @Description
+#! This function animates the <A>surface</A> using the configuration in <A>printRecord</A> and writes the animation to the <A>filename</A> into the working directory.
```

Comment:
How can this animation be watched? In which format is it stored?

The documentation of this method needs to be excellent since this is the first documentation that most people will see.